Content.Server/Disease/DiseaseSystem.cs

```diff
@@ -105,17 +106,29 @@ public override void Update(float frameTime)
                     var args = new DiseaseEffectArgs(carrierComp.Owner, disease, EntityManager);
                     disease.Accumulator -= disease.TickTime;
 
+                    int stage = 0; //defaults to stage 0 because you should always have one
+                    float lastTreshold = 0;
+                    foreach (var threshold in disease.Stages)
+                    {
+                        if (disease.TotalAccumulator >= threshold.Value &&
+                            threshold.Value > lastTreshold)
+                        {
+                            lastTreshold = threshold.Value;
+                            stage = threshold.Key;
+                        }
+                    }
```

Comment:
```suggestion
                    float lastThreshold = 0;
                    foreach (var threshold in disease.Stages)
                    {
                        if (disease.TotalAccumulator >= threshold.Value &&
                            threshold.Value > lastTreshold)
                        {
                            lastThreshold = threshold.Value;
                            stage = threshold.Key;
                        }
                    }
```

Also should the thresholds just be an array?